backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/codec/impl/LuceneByteVectorCodec.java

```diff
@@ -6,17 +6,64 @@
  */
 package org.hibernate.search.backend.lucene.types.codec.impl;
 
+import static org.apache.lucene.util.VectorUtil.cosine;
+import static org.apache.lucene.util.VectorUtil.dotProduct;
+
+import java.lang.invoke.MethodHandles;
+import java.util.function.Function;
+
+import org.hibernate.search.backend.lucene.logging.impl.Log;
 import org.hibernate.search.backend.lucene.lowlevel.codec.impl.HibernateSearchKnnVectorsFormat;
+import org.hibernate.search.util.common.AssertionFailure;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
 
 import org.apache.lucene.document.KnnByteVectorField;
 import org.apache.lucene.index.IndexableField;
 import org.apache.lucene.index.VectorEncoding;
 import org.apache.lucene.index.VectorSimilarityFunction;
 
 public class LuceneByteVectorCodec extends AbstractLuceneVectorFieldCodec<byte[]> {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private final Function<byte[], byte[]> normalizer;
+
 	public LuceneByteVectorCodec(VectorSimilarityFunction vectorSimilarity, int dimension, Storage storage, Indexing indexing,
-			byte[] indexNullAsValue, HibernateSearchKnnVectorsFormat knnVectorsFormat) {
+			byte[] indexNullAsValue, Float magnitude, HibernateSearchKnnVectorsFormat knnVectorsFormat) {
 		super( vectorSimilarity, dimension, storage, indexing, indexNullAsValue, knnVectorsFormat );
+		if ( magnitude != null ) {
+			normalizer = vector -> normalize( vector, magnitude );
+		}
+		else {
+			normalizer = this::normalizeNoop;
+		}
+	}
+
+	private byte[] normalizeNoop(byte[] vector) {
+		// do nothing
+		return vector;
+	}
+
+	private byte[] normalize(byte[] vector, float magnitude) {
+		float norm = norm( vector, vectorSimilarity );
+		if ( Math.abs( norm - magnitude ) > EPS ) {
+			throw log.byteVectorLengthExpectationFailed( vector, magnitude, norm );
+		}
+		return vector;
+	}
+
+	private static float norm(byte[] vector, VectorSimilarityFunction similarityFunction) {
+		switch ( similarityFunction ) {
+			case EUCLIDEAN:
+				return (float) Math.sqrt( dotProduct( vector, vector ) );
+			case MAXIMUM_INNER_PRODUCT:
+			case DOT_PRODUCT:
+				return dotProduct( vector, vector );
+			case COSINE:
+				return cosine( vector, vector );
+			default:
+				throw new AssertionFailure( "Unknown similarity function: " + similarityFunction );
+		}
```

Comment:
to double-check and if all ok then create our own utils, since these are currently from `org.apache.lucene.util.VectorUtil` ...

Reply:
Did you check that we needed client-side normalization/checks for Elasticsearch/OpenSearch? If recent versions of Elasticsearch/OpenSearch implement normalization/checks themselves, I wouldn't bother.